### PR TITLE
document difference between solvers + new solver normalization

### PR DESCRIPTION
**Pull request by lcnr**
while writing this doc I realized some further improvements to normalization.

Also, this existing system is far more involved than I expected as large parts of it are intuition based for me. Will try to play a bit with our approach in the near future and then improve this guide further.

cc @rust-lang/initiative-trait-system-refactor

### CODE REVIEW

File: src/solve/significant-changes.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+### Deferred alias equality
+
+The new implementation emits `AliasRelate` goals when relating aliases while the
+old implementation structurally relates the aliases instead. This enables the
+new solver to stall equality until it is able to normalize the related aliases.
+
+The behavior of the old solver is incomplete and relies on eager normalization
+which replaces ambiguous aliases with inference variables. As this is not
+not possible for aliases containing bound variables, the old implementation does
+nto handle aliases inside of binders correctly, e.g. [#102048]. See the chapter on
```
Comment (oli-obk) [inline src/solve/significant-changes.md:47]:
```suggestion
not handle aliases inside of binders correctly, e.g. [#102048]. See the chapter on
```

File: src/solve/significant-changes.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+As the new implementation has to be able to eagerly handle nested goals for
+candidate selection, always doing so reduces complexity. It may also enable
+us to merge more candidates in the future.
+
+[eval-nested]: https://github.com/rust-lang/rust/blob/master/compiler/rustc_trait_selection/src/traits/select/mod.rs#L1271-L1277
+[fulfill-nested]: https://github.com/rust-lang/rust/blob/df8ac8f1d74cffb96a93ae702d16e224f5b9ee8c/compiler/rustc_trait_selection/src/traits/fulfill.rs#L708-L712
+
+### Nested goals are evaluated until reaching a fixpoint
+
+The new implementation always evaluates goals in a loop until reachin a fixpoint.
```
Comment (oli-obk) [inline src/solve/significant-changes.md:68]:
```suggestion
The new implementation always evaluates goals in a loop until reaching a fixpoint.
```

File: src/solve/normalization.md [hunk 3]
```diff
@@ -3,32 +3,80 @@
...
+We now differentiate between "one-step normalization", "structural normalization" and
+"deep normalization".
+
+## One-step normalization
+
+One-step normalization is implemented via `NormalizesTo` goals. Unlike other goals
+in the trait solver, `NormalizesTo` always expects the term to be an unconstrained
+inference variable[^opaques]. Think of it as a function, taking an alias as input
+and returning its underlying value. If the alias is rigid, `NormalizesTo` fails and
+returns `NoSolution`.
```
Comment (oli-obk) [inline src/solve/normalization.md:15]: when does it happen that the alias is rigid?

Comment (lcnr) [reply src/solve/normalization.md]: extended that section to explain taht

**Review by BoxyUwU — APPROVED** [review]

**Pull request merged**